test/handler.test.js

```diff
@@ -176,14 +172,39 @@ describe('Handlers [index.js]', () => {
     })
 
     it('should return data fields', () => {
+      testData.data = {
+        url: 'http://www.gutenberg.org/ebooks/10.epub.images',
+        id: '10',
+        updated: moment().format(),
+      }
       testRecord.kinesis.data = Buffer.from(JSON.stringify(testData)).toString('base64')
       const results = Lambda.readFromKinesis(testRecord.kinesis.data)
       expect(results[0]).to.equal('http://www.gutenberg.org/ebooks/10.epub.images')
       expect(results[1]).to.equal('10')
       expect(results[2]).to.deep.equal(new Date(testData.data.updated))
+      expect(results[3]).to.equal('10_images.epub')
+    })
+
+    it('should should transform 00000.epub.(no)images URLs', () => {
+      testData.data = {
+        url: 'http://www.gutenberg.org/ebooks/9999.epub.noimages',
+        id: '9999',
+        updated: moment().format(),
+      }
+      testRecord.kinesis.data = Buffer.from(JSON.stringify(testData)).toString('base64')
+      const results = Lambda.readFromKinesis(testRecord.kinesis.data)
+      expect(results[0]).to.equal('http://www.gutenberg.org/ebooks/9999.epub.noimages')
+      expect(results[1]).to.equal('9999')
+      expect(results[2]).to.deep.equal(new Date(testData.data.updated))
+      expect(results[3]).to.equal('9999_noimages.epub')
     })
 
     it('should throw LambdaError if regex match fails', () => {
+      testData.data = {
+        url: 'http://www.gutenberg.org/ebooks/bad.epub.images',
```

Comment:
Why would the regex match fail on this?

Reply:
Thanks for catching this, I'm not sure why I had the test that way. You're correct that it should just start with the `/notReal` URL. (And you're right that the `bad.epub.images` would actually be a valid URL)